build_runner_core/lib/src/environment/build_environment.dart

```diff
@@ -40,11 +40,18 @@ abstract class BuildEnvironment {
   /// completes, it will expire afterwords since it can no longer guarantee a
   /// consistent state.
   ///
+  /// The [outputMap] is a map of input paths to a list of output paths. If the
+  /// input path is an empty string it means all input paths. If the list of
+  /// output paths is empty it means only output to the default location.
```

Comment:
What does "default location" mean here? If the paths are empty there won't be any output in the sense that we use the term elsewhere. We don't consider the generated directory under `.dart_tool` to be an output.